src/main/java/bio/terra/janitor/service/cleanup/FlightManager.java

```diff
@@ -57,14 +60,18 @@ public FlightManager(
    * to be submitted to Stairway.
    */
   public Optional<String> submitFlight(Instant expiredBy) {
+    Stopwatch stopwatch = Stopwatch.createStarted();
     String flightId = stairway.createFlightId();
     Optional<TrackedResource> resource = updateResourceForCleaning(expiredBy, flightId);
     if (!resource.isPresent()) {
       // No resource to schedule.
       return Optional.empty();
     }
     // If submission fails, it will be recovered later.
-    submitToStairway(flightId, resource.get());
+    boolean submissionSuccessful = submitToStairway(flightId, resource.get());
+    // Only record duration of submission if there was something to attempt to schedule.
+    MetricsHelper.recordSubmissionDuration(
```

Comment:
Currently we only record the duration and the boolean. Is that enough? Do we want more information like the flight id, resource type, client id stuff for easier debug?

Reply:
These initial metrics are for how long the interaction between the JanitorDao and Stairway take, so I thought to only include what might be interesting there. I don't think the resource type or the client id will matter for those (and there's a limit on how many distinct values you should have in your metrics, so we really shouldn't include the flight id which is always unique. flight id is more for logs).
We'll still want to add more metrics about the resource type, client id, and tracked resource state, but not in this PR. I'm thinking we periodically query the database with some select-count-groupby and export that.

Reply:
Just wanted to add a +1 to what Will is saying here. When choosing labels to associate with a metric, it's really important to avoid using anything we know will exhibit high cardinality such as short-lived unique ids. This can cause performance issues when querying metrics. This can also cause the number of unique time series tracked in stackdriver to massively increase which could lead to quota/billing issues. So I think this is a good thing to keep in mind when selecting metrics labels.